core/services/gateway/handlers/functions/handler.functions.go

```diff
@@ -48,10 +104,21 @@ func (h *functionsHandler) HandleNodeMessage(ctx context.Context, msg *api.Messa
 	return nil
 }
 
-func (h *functionsHandler) Start(context.Context) error {
+func (h *functionsHandler) Start(ctx context.Context) error {
+	if h.allowlist != nil {
```

Comment:
Should not this handler implement StartStopOnce to prevent such the checks?

Reply:
Allowlist can be nil if AllowlistCheckEnabled is set to false. But good point about StertSTopOnce, I'll add it (maybe in a separate PR).